### PR TITLE
ansible-lint: Fix jinja[invalid], no-jinja-when, yaml[line-legnth], package-latest failures

### PR DESCRIPTION
**Pull request by liangwen12year**

### CODE REVIEW

**Comment by liangwen12year** [comment]
[citest]

File: tests/playbooks/tests_ethtool_features.yml [hunk 1]
```diff
@@ -134,6 +134,7 @@
                 msg: "##################################################"
             - network_connections:
                 provider: "{{ network_provider | mandatory }}"
+                # noqa jinja[invalid]
```
Comment (ffmancera) [inline tests/playbooks/tests_ethtool_features.yml:137]: Same here.

**Review by ffmancera — CHANGES_REQUESTED** [review]

File: tasks/main.yml [hunk 2]
```diff
@@ -115,7 +115,7 @@
 
 - name: Configure networking connection profiles
   network_connections:
-    provider: "{{ network_provider | mandatory }}"
+    provider: "{{ network_provider | mandatory }}"  # noqa jinja[invalid]
```
[the remark this answers is not in the document]
Comment (richm) [reply tasks/main.yml:118]: What is the error here?  This looks like valid jinja to me?

[the remark this answers is not in the document]
Comment (liangwen12year) [reply tasks/main.yml:118]: The error is:

```
jinja[invalid]: Mandatory variable 'network_provider'  not defined.
tasks/main.yml:118 Task/Handler: Configure networking connection profiles
```

[the remark this answers is not in the document]
Comment (ffmancera) [reply tasks/main.yml]: It seems the jinja cannot be removed here. Then we have two different solutions:
1. Redefine the variable so it doesn't fail
2. set #noqa as you did before

What do you think? @liangwen12year

[the remark this answers is not in the document]
Comment (richm) [reply tasks/main.yml]: Ah, I see.  https://ansible-lint.readthedocs.io/configuring/#ansible-lint-configuration
```
# Define required Ansible's variables to satisfy syntax check
extra_vars:
  foo: bar
```
so add an extra_vars for `network_provider: nm` - it shouldn't matter what the value is, only that it is defined to something reasonable

[the remark this answers is not in the document]
Comment (liangwen12year) [reply tasks/main.yml]: > It seems the jinja cannot be removed here. Then we have two different solutions:
> 
> 1. Redefine the variable so it doesn't fail
> 2. set #noqa as you did before
> 
> What do you think? @liangwen12year

I prefer defining it as the extra var as @richm suggested.

**Comment by liangwen12year** [comment]
[citest]

File: .ansible-lint [hunk 3]
```diff
@@ -2,6 +2,7 @@
 profile: production
 extra_vars:
   test_playbook: tests_default.yml
+  network_provider: nm
```
Comment (tyll) [inline .ansible-lint:5]: This change needs a commit message that explains the reason for this and this should be a separate commit.

Comment (liangwen12year) [reply .ansible-lint:5]: I separated it into another commit, Thanks.

**Review by ffmancera — CHANGES_REQUESTED** [review]
Since you introduced a new commit for `jinja[invalid]` the commit ` ansible-lint: Fix jinja[invalid], no-jinja-when, yaml[line-legnth], package-latest failures` should not mention `jinja[invalid]`. Other than that LGTM

**Comment by liangwen12year** [comment]
> Since you introduced a new commit for `jinja[invalid]` the commit ` ansible-lint: Fix jinja[invalid], no-jinja-when, yaml[line-legnth], package-latest failures` should not mention `jinja[invalid]`. Other than that LGTM

I updated it. Thanks.

**Review by ffmancera — APPROVED** [review]
